traffic_ops/traffic_ops_golang/cachegroup/queueupdate.go

```diff
@@ -0,0 +1,164 @@
+package cachegroup
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/apache/incubator-trafficcontrol/lib/go-log"
+	"github.com/apache/incubator-trafficcontrol/lib/go-tc"
+	"github.com/apache/incubator-trafficcontrol/traffic_ops/traffic_ops_golang/api"
+	"github.com/apache/incubator-trafficcontrol/traffic_ops/traffic_ops_golang/auth"
+)
+
+type QueueUpdatesReq struct {
+	Action string      `json:"action"`
+	CDN    *tc.CDNName `json:"cdn"`
+	CDNID  *string     `json:"cdnId"`
```

Comment:
On the perl side an int or string works, I think int is the type we want moving forward, with support for a string for backwards compatibility.

Reply:
I agree int is better, but it's documented as being a string.

Normally I don't like making the API support undocumented things, it leads to difficult to support, maintain, and understand clients and servers. But in this case, it seems likely many clients are passing the ID as an int.

 I'll change it to decode either.